packages/core/core/src/AssetGraphBuilder.js

```diff
@@ -71,6 +73,10 @@ export default class AssetGraphBuilder extends EventEmitter {
         }
       });
     }
+
+    if (opts.cliOpts.hot) {
+      this.hmrServer = new HMRServer(this, opts.cliOpts);
```

Comment:
I am wondering about potentially keeping the HMRServer completely separate from the AssetGraphBuilder, and keep that just for building the graph. Possibly it could live in Parcel instead, which is the glue code between all the pieces?